packages/tldraw/src/lib/shapes/shared/TextLabel.tsx

```diff
@@ -71,48 +74,59 @@ export const TextLabel = React.memo(function TextLabel<
 			style={{
 				justifyContent: align === 'middle' || legacyAlign ? 'center' : align,
 				alignItems: verticalAlign === 'middle' ? 'center' : verticalAlign,
+				...(bounds
+					? {
+							top: bounds.minY,
+							left: bounds.minX,
+							width: bounds.width,
+							height: bounds.height,
+							position: 'absolute',
+					  }
+					: {}),
```

Comment:
added the ability to use the shape's actual bounds as the text label placement, rather than relying on the assumption that the bounds start at (0, 0)